defdispjlib.py

```diff
@@ -87,24 +87,162 @@ def parse_deformable_displacement(rw, ed):
         parse_rotmat(rw, 19, R2)
         el.rotoffsets[1].value = R2.to_quaternion();
 
-        el.import_function = "add.mbdyn_elem_deformable_displacement"
+        el.import_function = "add.mbdyn_elem_defdisp"
+        el.update = "update_defdisp"
         el.name = el.type + "_" + str(el.int_label)
         el.is_imported = True
         ret_val = False
     return ret_val
 # -----------------------------------------------------------
-# end of parse_deformable_displacement(rw, ed) function
+# end of parse_defdisp(rw, ed) function
+
+def update_defdisp(elem, insert_keyframe = False):
+
+    nd = bpy.context.scene.mbdyn.nodes
+
+    defdispOBJ = bpy.data.objects[elem.blender_object]
+    n1 = nd['node_' + str(elem.nodes[0].int_label)].blender_object
+    n2 = nd['node_' + str(elem.nodes[1].int_label)].blender_object
+
+    # nodes' objects
+    n1OBJ = bpy.data.objects[n1]
+    n2OBJ = bpy.data.objects[n2]
+
+    # get offsets
+    f1 = elem.offsets[0].value
+    f2 = elem.offsets[1].value
+    q1 = elem.rotoffsets[0].value
+    q2 = elem.rotoffsets[1].value
+
+    # assign coordinates of knots in global frame
+    R1 = n1OBJ.rotation_quaternion.to_matrix()
+    R2 = n2OBJ.rotation_quaternion.to_matrix()
+    p1 = n1OBJ.location + R1 * Vector((f1[0], f1[1], f1[2]))
+    p2 = n2OBJ.location + R2 * Vector((f2[0], f2[1], f2[2]))
+
+    R1h =  Quaternion(( q1[0], q1[1], q1[2], q1[3] )).to_matrix()
+
+    defdispOBJ.location = (p2 + p1)/2
+    def_vector = (p2 - p1)
+    def_vector = R1h.transposed() * R1.transposed() * def_vector
+    print(def_vector)
```

Comment:
@zanoni-mbdyn Here is the math in the update method. I think its follows your instructions.